packages/block-library/src/index.js

```diff
@@ -38,6 +38,7 @@ import * as navigation from './navigation';
 import * as navigationLink from './navigation-link';
 import * as latestComments from './latest-comments';
 import * as latestPosts from './latest-posts';
+import * as legacyWidget from './legacy-widget';
```

Comment:
I still think this should only be available in the widgets editor :)

Reply:
I moved it back to `block-library` in this PR so that I'd be forced to, while rewriting Legacy Widget, to not write any code that relies on `edit-navigation`. It would be straightforward to move it back to `edit-navigation`. 

I'm coming around to it living in `block-library`, though, as I think it would be a very useful block to have in the Site Editor. I can imagine a scenario where I'm using a FSE theme but want to plop in a widget that hasn't yet been made into a block.